plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-payment-gateways-controller.php

```diff
@@ -223,4 +223,52 @@ public function get_item_schema() {
 
 		return $this->add_additional_fields_schema( $schema );
 	}
+
+	/**
+	 * Validate multiselect based settings (with support for nested options).
+	 *
+	 * @param array|string $values  The submitted values.
+	 * @param array        $setting The field settings.
+	 * @return array|WP_Error
+	 */
+	public function validate_setting_multiselect_field( $values, $setting ) {
```

Comment:
This fixes an issue where the validation of options keys wasn't working as expected (because the array was multidirectional, but it was expecting a flat array)